contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/fixture/invoicing/InvoicingApplicationTests.java

```diff
@@ -476,20 +631,106 @@ void putJson_shouldReturn_http204() throws Exception {
+                                    .accept(MediaType.APPLICATION_JSON)
+                                    .header(HttpHeaders.IF_MATCH, toEtag(prevVersion))
+                                    .content("""
+                                            {
+                                                "_links": {
+                                                    "customer" : {
+                                                        "href": "/customers/%s"
+                                                    }
+                                                }
+                                            }
+                                            """.formatted(correctCustomerId)))
+                            .andExpect(status().isNoContent());
+                    var invoice = invoices.findById(invoiceId(INVOICE_NUMBER_1)).orElseThrow();
+                    assertThat(invoice.getVersion()).isNotEqualTo(prevVersion);
+                }
+
+                @Test
+                @Disabled("ACC-1188")
+                void putJson_withIfBadMatch_http412() throws Exception {
+                    // fictive example: fix the customer
+                    var correctCustomerId = customers.findByVat(ORG_INBEV_VAT).orElseThrow().getId();
+                    var prevVersion = invoices.findById(invoiceId(INVOICE_NUMBER_1)).orElseThrow().getVersion();
+                    mockMvc.perform(put("/invoices/" + invoiceId(INVOICE_NUMBER_1) + "/counterparty")
+                                    .contentType(MediaType.APPLICATION_JSON)
+                                    .accept(MediaType.APPLICATION_JSON)
+                                    .header(HttpHeaders.IF_MATCH, "\"INVALID\"")
+                                    .content("""
+                                            {
+                                                "_links": {
+                                                    "customer" : {
+                                                        "href": "/customers/%s"
+                                                    }
+                                                }
+                                            }
+                                            """.formatted(correctCustomerId)))
+                            .andExpect(status().isPreconditionFailed());
+                    var invoice = invoices.findById(invoiceId(INVOICE_NUMBER_1)).orElseThrow();
+                    assertThat(invoice.getVersion()).isEqualTo(prevVersion);
+                }
+
             }
 
             @Nested
             class OneToMany {
 
                 @Test
                 void putJson_shouldReplaceLinksAndReturn_http204_noContent() throws Exception {
-                    var xenit = customers.findByVat(ORG_XENIT_VAT).orElseThrow();
-                    var newOrderId = orders.save(new Order(xenit)).getId();
+                    AtomicReference<UUID> newOrderId = new AtomicReference<>(UUID.randomUUID());
```

Comment:
```suggestion
                    AtomicReference<UUID> newOrderId = new AtomicReference<>(null);
```

Don't set a random ID here.
Alternatively, create an overload for `doInTransaction()` which does return a value